pkg/cliplugins/workspace/cmd/workspace.go

```diff
@@ -0,0 +1,570 @@
+	scope, _ := getScopeAndName(workspaceConfigCurrentContext)
+	workspaceContextName := kcpWorkspaceContextNamePrefix + workspaceConfigCurrentContext
+
+	currentContextName := o.startingConfig.CurrentContext
+	if o.kubectlOverrides.CurrentContext != "" {
+		currentContextName = o.kubectlOverrides.CurrentContext
+	}
+
+	currentContext := o.startingConfig.Contexts[currentContextName]
+	var currentContextAuthInfo *api.AuthInfo
+	if currentContext != nil {
+		currentContextAuthInfo = o.startingConfig.AuthInfos[currentContext.AuthInfo]
+	}
+	o.startingConfig.Clusters[workspaceContextName] = workspaceConfig.Clusters[workspaceConfigCurrentContext]
+	workspaceAuthInfo := PrioritizedAuthInfo(&o.kubectlOverrides.AuthInfo, currentContextAuthInfo)
+	o.startingConfig.AuthInfos[workspaceContextName] = workspaceAuthInfo
+
+	o.startingConfig.Contexts[workspaceContextName] = &api.Context{
+		Cluster:  workspaceContextName,
+		AuthInfo: workspaceContextName,
+	}
+
+	o.startingConfig.CurrentContext = workspaceContextName
+
+	if _, err := o.Out.Write([]byte(fmt.Sprintf("Current %s workspace is \"%s\".\n", scope, workspaceName))); err != nil {
+		return err
+	}
+	return clientcmd.ModifyConfig(o.configAccess, *o.startingConfig, true)
+}
+
+func (o *WorkspaceOptions) getCurrentWorkspace() (scope string, name string, err error) {
+	currentContextName := o.startingConfig.CurrentContext
+	if o.kubectlOverrides.CurrentContext != "" {
+		currentContextName = o.kubectlOverrides.CurrentContext
+	}
+
+	if currentContextName == "" {
+		return "", "", errors.New("No current context !")
+	}
+
+	if !strings.HasPrefix(currentContextName, kcpWorkspaceContextNamePrefix) {
+		return "", "", errors.New("The current context is not a KCP workspace !")
+	}
+
+	scope, workspaceName := getScopeAndName(strings.TrimPrefix(currentContextName, kcpWorkspaceContextNamePrefix))
+	return scope, workspaceName, nil
+}
+
+func (o *WorkspaceOptions) checkCurrentWorkspace(workspaceName string, tenancyClient *tenancyclient.TenancyV1alpha1Client) (err error) {
+	if _, err := tenancyClient.Workspaces().Get(context.TODO(), workspaceName, metav1.GetOptions{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *WorkspaceOptions) RunCurrent() error {
+	workspaceDirectoryRestConfig, err := o.WorkspaceDirectoryRestConfig()
+	if err != nil {
+		return err
+	}
+
+	tenancyClient, err := tenancyclient.NewForConfig(workspaceDirectoryRestConfig)
+	if err != nil {
+		return err
+	}
+
+	scope, workspaceName, err := o.getCurrentWorkspace()
+	outputCurrentWorkspaceMessage := func() error {
+		if workspaceName != "" {
+			_, err := o.Out.Write([]byte(fmt.Sprintf("Current %s workspace is \"%s\".\n", scope, workspaceName)))
+			return err
+		}
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// Check that the scope is consistent with the workspace-directory config
+	if !strings.HasSuffix(workspaceDirectoryRestConfig.Host, "/"+scope) {
+		_ = outputCurrentWorkspaceMessage()
```

Comment:
Is this a side-effect call?

Reply:
No, it simply writes to the output of the CLI.

Reply:
That's a side-effect :)